File: Sorting/In C/SelectionSort.c
//Selection Sort

#include<stdio.h>
#include<conio.h>

void getArray(int array[],int n){

    for(int i=0;i<n;i++){
        scanf("%d",&array[i]);
    }
}

void dispArray(int array[],int n){

    for(int i=0;i<n;i++){
        printf("%d ",array[i]);
    }   

    // debugging printf("\n");
}

void SelectionSort(int array[],int n){
    int min;
    int temp;
    for(int i=0;i<n;i++){
         min=i;
         for(int j=i;j<n;j++){
            if(array[j]<array[min]){
                min=j;
            }
         }

        temp=array[min];
        array[min]=array[i];
        array[i]=temp;

    }}

int main(){

    int n,array[50],key,r;
    printf("Enter number of elements in the array: ");
    scanf("%d",&n);
    getArray(array,n);

    printf("The given array is ");
    dispArray(array,n);
    SelectionSort(array,n);
    printf("\nThe Sorted array is ");
    dispArray(array,n);

}

File: Sorting/In Python #prefer/SelectionSort.py
# Selection sort in Python

def selectionSort(array, size):
   
    for step in range(size):
        min_idx = step

        for i in range(step + 1, size):
         
            # to sort in descending order, change > to < in this line
            # select the minimum element in each loop
            if array[i] < array[min_idx]:
                min_idx = i
         
        # put min at the correct position
        (array [step], array[min_idx]) = (array[min_idx], array[step])

def getList(array,n):
    for i in range(n):
        print("Enter value {} :".format(i+1))
        val=int(input())
        array.append(val)

n=int(input("\nEnter the number of elements in the list:"))

print("\nEnter the list elements:")
array=[]
getList(array,n)

selectionSort(array,n)

print('Sorted Array in Ascending Order:')
print(array)